finetune/input_pipeline.py

```diff
@@ -209,13 +198,19 @@ def _format_for_pipeline(data):
             data_ = lambda: itertools.repeat(None)
         return data_
         
-    def _dataset_without_targets(self, Xs, train, context=None, update_hook=None, forced_mask=None):
-        Xs_ = self._format_for_pipeline(Xs)
-        ys_ = self._format_for_pipeline(None)
-        context_ = self._format_for_pipeline(context)
-        forced_mask_ = self._format_for_pipeline(forced_mask)
+    def _dataset_without_targets(self, Xs, train, context=None, update_hook=None, forced_mask=None, from_generator=False):
+        if from_generator:
+            assert forced_mask is None, "Forced mask is not supported with generator inputs"
+            def Xs_gen():
+                for x in Xs():
+                    yield (x.get("X"), x.get("y"), x.get("context"), None)
```

Comment:
we could get into a strange situation if 'y' was actually provided for some reason

Reply:
Can we just replace this with a None then?